haskell/private/actions/repl.bzl

```diff
@@ -133,6 +140,37 @@ def build_haskell_repl(
         is_executable = True,
     )
 
+    ghc_info = {
+        "has_version": "true" if pkg_ghc_info["has_version"] else "false",
+        "library_path": library_deps,
+        "ld_library_path": ld_library_deps,
+        "packages": pkg_ghc_info["packages"],
+        "package_ids": pkg_ghc_info["package_ids"],
+        "package_dbs": pkg_ghc_info["package_dbs"],
+        "lib_imports": lib_imports,
+        "libraries": libraries,
+        "lib_sources": lib_sources,
+        "bin_sources": bin_sources,
+        "execs": {
+            "ghc": hs.tools.ghc.path,
+            "ghci": hs.tools.ghci.path,
+            "runghc": hs.tools.runghc.path,
+        },
+        "flags": {
+            "compiler": compiler_flags,
+            "toolchain_compiler": hs.toolchain.compiler_flags,
+            "repl": repl_ghci_args,
+            "toolchain_repl": hs.toolchain.repl_ghci_args,
+        },
+    }
+    ghc_info_file = hs.actions.declare_file(
+        target_unique_name(hs, "ghc-info"),
+    )
+    hs.actions.write(
+        output = ghc_info_file,
+        content = str(ghc_info),
```

Comment:
Ceci n’est pas une JSON string. :smile: 
Instead of a `dict`, you need to use a `struct` (which is a `dict` with immutable fields) and use its [`to_json()` function](https://docs.bazel.build/versions/master/skylark/lib/struct.html#to_json).

Reply:
Yeap, just found about this an hour ago. Will fix!